### PR TITLE
Add --worker-port-list option to ray start

### LINKED ISSUE #11203: [core] Support providing a list of worker ports
### Describe your feature request
Right now, we support port selection by providing a range of port numbers to use. 

However, for @tgaddair's use case, we will need to support providing a list of worker ports (which may not necessarily be contiguous).

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Allows users to specify a list of specific ports to use for workers. Takes precedence over min-worker-port and max-worker-port.

## Related issue number

Closes https://github.com/ray-project/ray/issues/11203

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@richardliaw I did manual testing and added a very simple test to `test_multi_node`, but couldn't think of a straightforward way to do more complete testing on travis. LMK if you have any ideas.

File: python/ray/parameter.py [hunk 1]
```diff
@@ -252,6 +257,20 @@ def update_if_absent(self, **kwargs):
...
 
     def _check_usage(self):
+        if self.worker_port_list is not None:
+            for port_str in self.worker_port_list.split(","):
+                try:
+                    port = int(port_str)
+                except ValueError as e:
+                    raise ValueError(
+                        "worker_port_list must be a comma-separated " +
+                        "list of integers: {}".format(e))
```
Comment (richardliaw) [inline python/ray/parameter.py:267]:
```
In [3]:                 try:
   ...:                     port = int(port_str)
   ...:                 except ValueError as e:
   ...:                     raise ValueError(
   ...:                         "worker_port_list must be a comma-separated " +
   ...:                         "list of integers: {}".format(e))
   ...:
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-160cfabed35e> in <module>
      1 try:
----> 2     port = int(port_str)
      3 except ValueError as e:

ValueError: invalid literal for int() with base 10: 'hi'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-3-160cfabed35e> in <module>
      4     raise ValueError(
      5         "worker_port_list must be a comma-separated " +
----> 6         "list of integers: {}".format(e))
      7

ValueError: worker_port_list must be a comma-separated list of integers: invalid literal for int() with base 10: 'hi'
```

Comment (richardliaw) [reply python/ray/parameter.py:267]: To avoid unnecessary exception chaining, maybe do `raise ValueError(...) from None`

**Review by richardliaw — APPROVED** [review]
Looks good, just small nit about exceptions

File: python/ray/parameter.py (hunk 1 above)
Comment (edoakes) [reply python/ray/parameter.py:267]: wow, I had no idea you could do this! very cool

**Comment by richardliaw** [comment]
fyi the tests are all failing for some reason

**Pull request merged**